File: leetcode/Stack/0071_Simplify_Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        dirs = path.split('/')
        stack = []
        for dir in dirs:
            if not dir or dir == '.':
                pass
            elif dir == '..':
                if stack:
                    stack.pop()
            else:
                stack.append(dir)

        return '/' + '/'.join(stack)

File: leetcode/Stack/0071_Simplify_Path.cpp
class Solution {
public:
	string simplifyPath(string path) {
		if (path.empty())
			return "";

		vector<string> level;
		string temp = "";

		path += '/';
		for (char c : path) {
			if (c == '/') {
				if (temp == ".." && !level.empty())
					level.pop_back();
				else if (!temp.empty() && temp != "." && temp != "..")
					level.emplace_back(temp);
				temp.clear();
			}
			else
				temp += c;
		}

		if (level.empty())
			return "/";
		else {
			string result = "";
			for (string s : level)
				result += ("/" + s);
			return result;
		}
	}
};